From d4ff4f46fc341b7941362f8b4f8288eef53ad3e7 Mon Sep 17 00:00:00 2001
From: Koji Matsuoka <koji.matsuoka.xm@renesas.com>
Date: Mon, 23 Oct 2017 12:53:14 +0900
Subject: [PATCH 279/909] drm/brdige: dumb-vga-dac: Add connector status
 setting option

commit 1c6300530e805a1f1386189665a2dbbe2a07a52f from
git://git.kernel.org/pub/scm/linux/kernel/git/horms/renesas-bsp.git

If EDID can not be read, this patch adds an option to set the
connector status to connector_status_connected.

Signed-off-by: Koji Matsuoka <koji.matsuoka.xm@renesas.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/bridge/dumb-vga-dac.c | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/drivers/gpu/drm/bridge/dumb-vga-dac.c b/drivers/gpu/drm/bridge/dumb-vga-dac.c
index 9837c8d69e69..0894939a8966 100644
--- a/drivers/gpu/drm/bridge/dumb-vga-dac.c
+++ b/drivers/gpu/drm/bridge/dumb-vga-dac.c
@@ -1,6 +1,7 @@
 /*
  * Copyright (C) 2015-2016 Free Electrons
  * Copyright (C) 2015-2016 NextThing Co
+ * Copyright (C) 2017 Renesas Electronics Corporation
  *
  * Maxime Ripard <maxime.ripard@free-electrons.com>
  *
@@ -26,6 +27,7 @@ struct dumb_vga {
 
 	struct i2c_adapter	*ddc;
 	struct regulator	*vdd;
+	bool			ddc_flag;
 };
 
 static inline struct dumb_vga *
@@ -82,6 +84,13 @@ dumb_vga_connector_detect(struct drm_connector *connector, bool force)
 {
 	struct dumb_vga *vga = drm_connector_to_dumb_vga(connector);
 
+	/*
+	 * Set connector_status_connected forcibly when EDID is not used
+	 * by option.
+	 */
+	if (!vga->ddc_flag)
+		return connector_status_connected;
+
 	/*
 	 * Even if we have an I2C bus, we can't assume that the cable
 	 * is disconnected if drm_probe_ddc fails. Some cables don't
@@ -158,11 +167,17 @@ static struct i2c_adapter *dumb_vga_retrieve_ddc(struct device *dev)
 {
 	struct device_node *phandle, *remote;
 	struct i2c_adapter *ddc;
+	struct dumb_vga *vga = dev_get_drvdata(dev);
 
 	remote = of_graph_get_remote_node(dev->of_node, 1, -1);
 	if (!remote)
 		return ERR_PTR(-EINVAL);
 
+	if (of_find_property(remote, "no-use-ddc", NULL))
+		vga->ddc_flag = false;
+	else
+		vga->ddc_flag = true;
+
 	phandle = of_parse_phandle(remote, "ddc-i2c-bus", 0);
 	of_node_put(remote);
 	if (!phandle)
-- 
2.17.1

